File: MiniAirbnbBackend/src/main/java/com/airbnb/miniairbnb/service/PropertyService.java
package com.airbnb.miniairbnb.service;

import com.airbnb.miniairbnb.model.Property;
import com.airbnb.miniairbnb.model.User;

import java.util.List;
import java.util.Optional;

public interface PropertyService {
    Property createProperty(Property property, User host); //creeaza o proprietate noua (doar pt host sau admin)

    Property updateProperty(Long propertyId, Property propertyDetails, User currentUser); //Actuaalizeaza o proprietate (doar owner-ul sau ADMIN)

    void deleteProperty(Long propertyId, User currentUser); //sterge o proprietate (doar owner-ul sau ADMIN

    Optional<Property> findPropertyById(Long id);//gaseste proprietatea dupa id

    List<Property> findAllActiveProperties(); //gaseste toate proprietatile active (pt guest)

    List<Property> findPropertiesByHost(User host); //gaseste toate proprietatile unui host

    List<Property> findActivePropertiesByCity(String city); //filtrare proprietati active dupa oras

    List<Property> findActivePropertiesByCountry(String country); //filtrare proprietati active dupa tara

    boolean isPropertyOwner(Long propertyId, User user); //verifica daca utilizatorul este owner-ul proprietatii
}

File: mini-airbnb-frontend/src/services/propertyService.ts
import api from "./api";
import type { Property } from "../types";

export const propertyService = {
  //obtine toate proprietatile active
  getAllProperties: async (
    city?: string,
    country?: string
  ): Promise<Property[]> => {
    const params = new URLSearchParams();
    if (city) params.append("city", city);
    if (country) params.append("country", country);

    const response = await api.get<Property[]>(
      `/properties?${params.toString()}`
    );
    return response.data;
  },

  //obtine o proprietate dupa id
  getPropertyById: async (id: number): Promise<Property> => {
    const response = await api.get<Property>(`/properties/${id}`);
    return response.data;
  },

  //obtine proprietatile utilizatorului curent
  getMyProperties: async (): Promise<Property[]> => {
    const response = await api.get<Property[]>("/properties/my-properties");
    return response.data;
  },

  //creeaza o proprietate noua
  createProperty: async (
    property: Omit<Property, "id" | "host" | "createdAt" | "updatedAt">
  ): Promise<Property> => {
    const response = await api.post<Property>("/properties", property);
    return response.data;
  },

  //actualizeaza o proprietate
  updateProperty: async (
    id: number,
    property: Partial<Property>
  ): Promise<Property> => {
    const response = await api.put<Property>(`/properties/${id}`, property);
    return response.data;
  },

  //sterge o proprietate
  deleteProperty: async (id: number): Promise<void> => {
    await api.delete(`/properties/${id}`);
  },
};
